### PR TITLE
Update: Logical-and is counted in `complexity` rule (fixes #8535)

### LINKED ISSUE #8535: Complexity rule prefers and `&&` over or `||`
* **ESLint Version:** 3.19.0
* **Node Version:** 7.7.3
* **npm Version:** 4.5.0

**What parser (default, Babel-ESLint, etc.) are you using?** : babel-eslint

**Please show your full configuration:**

```
{
  "parser": "babel-eslint",
  "rules": {
    "complexity": [2, 1]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js

let foo, bar, baz, bang;

function check(foo, bar, baz, bang) {
	if (!bar || !bang) {
		baz = foo;
	}
}
function check2(foo, bar, baz, bang) {
	if (!(bar && bang)) {
		baz = foo;
	}
}
function check3(foo, bar, baz, bang) {
	if (bar && bang) {
		baz = foo;
	}
}
function check4(foo, bar, baz, bang) {
	if (!bar && !bang) {
		baz = foo;
	}
}
function check5(foo, bar, baz, bang) {
	if (bar || bang) {
		baz = foo;
	}
}

check(foo, bar, baz, bang);
check2(foo, bar, baz, bang);
check3(foo, bar, baz, bang);
check4(foo, bar, baz, bang);
check5(foo, bar, baz, bang);
```

**What did you expect to happen?**

All functions check to check5 should have the same complexity

**What actually happened? Please include the actual, raw output from ESLint.**

```
   3:1  error  Function 'check' has a complexity of 3   complexity
   8:1  error  Function 'check2' has a complexity of 2  complexity
  13:1  error  Function 'check3' has a complexity of 2  complexity
  18:1  error  Function 'check4' has a complexity of 2  complexity
  23:1  error  Function 'check5' has a complexity of 3  complexity
```

It seems that `||` has more complexity than `&&` in the current implementation, but I don't think that is true. All the `check` have a complexity of 3. (That is what http://jsmeter.info/axtsgb/1#results reports)

### PR DESCRIPTION
**Pull request by platinumazure**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

See #8535.

**What changes did you make? (Give an overview)**

Logical-and now counts the same as logical-or for increasing cyclomatic complexity in a function. (`a && b` is the same as `!(!a || !b)`, so either operator should increase complexity)

Since this change will result in strictly non-decreasing complexity, this will result in more warnings reported, hence "Update".

**Is there anything you'd like reviewers to focus on?**

Not really.

### CODE REVIEW

File: lib/rules/complexity.js [hunk 1]
```diff
@@ -133,9 +133,7 @@ module.exports = {
          * @private
          */
         function increaseLogicalComplexity(node) {
-
-            // Avoiding &&
-            if (node.operator === "||") {
+            if (node.operator === "&&" || node.operator === "||") {
```
Comment (not-an-aardvark) [inline lib/rules/complexity.js]: Could this `if` statement be removed?

Comment (platinumazure) [reply lib/rules/complexity.js]: In fact, I can remove the whole function!

**Comment by platinumazure** [comment]
@not-an-aardvark Please take another look, at your convenience. Thanks!

**Comment by platinumazure** [comment]
Closing/reopening to attempt another AppVeyor build.

**Comment by jking90** [comment]
This was a breaking change

**Comment by j-f1** [comment]
@jking90 ESLint’s [policy](https://github.com/eslint/eslint/blob/db431cbcac1c31f792d3763681efdd6414861fb4/README.md#semantic-versioning-policy) is that a minor release can break your build, and they recommend using `~` to specify the version you require, instead of the default `^`.

**Pull request merged**